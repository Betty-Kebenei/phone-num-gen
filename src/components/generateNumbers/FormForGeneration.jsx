import React from 'react';
import PropTypes from 'prop-types';

const FormForGeneration = (props) => {
  const { onChange, onSubmit, quantity } = props;

  return (
    <form onSubmit={onSubmit}>
      <div className="form-group">
        {/* eslint jsx-a11y/label-has-associated-control: ["error", { assert: "either" } ] */}
        <label htmlFor="quantity">How many numbers do you want to generate? </label>
        <br />
        <input
          id="quantity"
          type="number"
          name='quantity'
          value={quantity}
          onChange={onChange}
          min={10}
          max={1000}
          required/>
      </div>
      <input className="btn btn-primary" type="submit" value="Submit" />
    </form>
  )
}
FormForGeneration.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  quantity: PropTypes.number.isRequired
}
export default FormForGeneration;
