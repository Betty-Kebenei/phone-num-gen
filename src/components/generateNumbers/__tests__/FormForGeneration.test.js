import React from 'react';
import renderer from 'react-test-renderer';

import FormForGeneration from '../FormForGeneration';

test('header component renders correctly', () => {
  const component = renderer.create(<FormForGeneration />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
