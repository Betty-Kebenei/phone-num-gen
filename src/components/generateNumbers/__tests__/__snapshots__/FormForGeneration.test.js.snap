// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component renders correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <label
      htmlFor="quantity"
    >
      How many numbers do you want to generate? 
    </label>
    <br />
    <input
      id="quantity"
      max={1000}
      min={10}
      name="quantity"
      required={true}
      type="number"
    />
  </div>
  <input
    className="btn btn-primary"
    type="submit"
    value="Submit"
  />
</form>
`;
